CMakeLists.txt

```diff
@@ -371,21 +386,26 @@ endif(EXTERNAL_METIS_PATH)
 
 # Avoid exposing third-party symbols when using DGL as a library.
 if((NOT MSVC) AND (NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin"))
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wl,--exclude-libs,ALL")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wl,--exclude-libs,ALL")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--exclude-libs,ALL")
 endif()
 
-# Compile gpu_cache
+file(GLOB gpu_cache_src
+  third_party/HugeCTR/gpu_cache/src/nv_gpu_cache.cu
+)
+# Manually build gpu_cache because CMake always builds it as shared
 if(USE_CUDA)
-  # Manually build gpu_cache because CMake always builds it as shared
-  file(GLOB gpu_cache_src
-    third_party/HugeCTR/gpu_cache/src/nv_gpu_cache.cu
-  )
   cuda_add_library(gpu_cache STATIC ${gpu_cache_src})
   target_include_directories(gpu_cache PRIVATE "third_party/HugeCTR/gpu_cache/include")
   target_include_directories(dgl PRIVATE "third_party/HugeCTR/gpu_cache/include")
   list(APPEND DGL_LINKER_LIBS gpu_cache)
   message(STATUS "Build with HugeCTR GPU embedding cache.")
+elif(USE_ROCM)
+ set_source_files_properties(${gpu_cache_src} PROPERTIES LANGUAGE HIP)
+ add_library(gpu_cache STATIC ${gpu_cache_src})
```

Comment:
I'm aware we have a hip_add_library but I'm not sure if or when it is preferred over just add_library.